src/Components/Endpoints/test/TypeNameHashTest.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Security.Cryptography;
+
+namespace Microsoft.AspNetCore.Components.Endpoints;
+
+public class TypeNameHashTest
+{
+    // In these tests, we're mostly interested in checking that the hash function succeeds
+    // for any type with a valid name. We'll also do some basic sanity checking by ensuring
+    // that the string representation of the hash has the expected length.
+
+    // We currently use a hex-encoded SHA1 hash, so there should be two characters per byte
+    // of encoded data.
+    private const int ExpectedHashLength = SHA1.HashSizeInBytes * 2;
```

Comment:
@javiercn has mentioned in one of our conversations that we should also update from SHA1 to SHA256

Reply:
Yes, please.